localstack-core/localstack/services/kms/provider.py

```diff
@@ -992,7 +1000,10 @@ def decrypt(
         self._validate_key_state_not_pending_import(key)
 
         try:
-            plaintext = key.decrypt(ciphertext, encryption_context)
+            if not ciphertext:
+                plaintext = key.decrypt_rsa(ciphertext_blob)
+            else:
+                plaintext = key.decrypt(ciphertext or ciphertext_blob, encryption_context)
```

Comment:
We should be using the `ciphertext` to decrypt here: 

`plaintext = key.decrypt(ciphertext, encryption_context)`